truckersmp_cli/configfile.py

```diff
@@ -0,0 +1,44 @@
+"""
+Configuration file handler for truckersmp-cli main script.
+
+Licensed under MIT.
+"""
+
+import configparser
+import os
+
+
+class ConfigFile:
+    """Configuration file."""
+
+    def __init__(self, configfile):
+        """
+        Initialize ConfigFile object.
+
+        configfile: Path to configuration file
+        """
+        self._thirdparty_wait = 0
+        self._thirdparty_executables = []
+        parser = configparser.ConfigParser()
+        parser.read(configfile)
+        for sect in parser.sections():
+            # get data from valid thirdparty.* sections
+            if not sect.startswith("thirdparty.") or "executable" not in parser[sect]:
+                continue
+            try:
+                wait = int(parser[sect]["wait"])
+            except (KeyError, ValueError):
+                wait = 0  # invalid or missing
+            self._thirdparty_wait = max(wait, self._thirdparty_wait)
+            self._thirdparty_executables.append(
+                os.path.expanduser(parser[sect]["executable"]))
```

Comment:
I'm not so sure about this one as this only expands `~` and leaves the other variables untouched (`$XDG…`, `$HOME`).
There's `expandvars()` which handles these but leaves `~` untouched. So we would have to use both together but that's still isn't resolving unset variables.

I guess we should make absolute path mandatory here.

Reply:
> I guess we should make absolute paths mandatory here.

Do we need to check whether the given path is absolute path?

Even if `os.path.isabs(given_path)` returns `False`, Windows style absolute paths (e.g. `C:\Program Files (x86)\TrucksBook Client\TB Client.exe`) work.

```
# When the program is located in the game prefix
executable = C:\Program Files (x86)\TrucksBook Client\TB Client.exe
```

Regarding relative paths: What about treating them as paths relative to our data directory (`Dir.truckersmp_cli_data`, `$XDG_DATA_HOME/truckersmp-cli`)?

```
[thirdparty.program1]
# $XDG_DATA_HOME/truckersmp-cli/program1/program1.exe
executable = program1/program1.exe
```

* `os.path.isabs(given_path)` returns `True` -> Use the given path
* `os.path.isabs(given_path)` returns `False`:
    * The given path is Windows style absolute path -> Use the given path
    * Not Windows style absolute path -> `os.path.join(Dir.truckersmp_cli_data, given_path)`

Reply:
I wasn't even aware windows paths work here.

> * `os.path.isabs(given_path)` returns `True` -> Use the given path
> * `os.path.isabs(given_path)` returns `False`:
>   * The given path is Windows style absolute path -> Use the given path
>   * Not Windows style absolute path -> `os.path.join(Dir.truckersmp_cli_data, given_path)`

This seems like a good compromise.